### ISSUE TITLE
Fallout 4 hangs when started by MO (without f4se)

### ISSUE THREAD

**Issue by ogrotten** [TanninOne/modorganizer#500]
Is there somewhere or some way that I can force come kind of verbose mode so that I can get some kind of report?

The game will not start when initiated thru MO. It will start without MO. Selecting the game, then Run will appear to start loading the game, and at some point will hang without error message (with between 0 and 2% cpu usage and no change in memory use) until I click on the black screen at which time I get the windows default message for a hung program.

This started a few days ago with 203, and I have wound up with several versions of MO installed in several dirs in an effort to chase this down.

My original 202 worked without problem except for known and expected issues (bodyslide).

I originally upgraded 202 > 203 about 2 weeks ago. I duplicated the dir and copied the contents of the 203 zip over the 202 install. It worked without problem except for known and expected (bodyslide)

A few days ago I made [this post](https://github.com/TanninOne/modorganizer/issues/485) and as a result made a new install dir for straight 203. I worked past the known issues of 203. (https://github.com/TanninOne/modorganizer/issues/486, my post is down a ways)

Once I had MO working, the game wouldn't start. The only thing that had changed was the installation of mo. No other software or hardware had changed except for turning off & on my steam controller a few times. 

I performed many troubleshooting steps, I have redownloaded and reinstalled MO from zip, and have un/reinstalled the game. *Occasionally* the game will start, but it is very rare, 1 of 10 tries.

To be clear:
* The game hangs if started thru MO.
* The game runs if started thru steam or windows icon.
* Regardless of mods loaded 
* Regardless of profile or game ini config

I've just now noticed that creating new profiles tends to allow it to work for 1 or 2 starts then again not at all. I'm digging around on it for a bit. Will report any discovery.

**Comment by ogrotten** [comment]
I watch the process manager. I can tell when the game is going to fail by a couple of things

1. CPU usage peaks at < 80% before dipping. When the game is going to succeed, it will peak at 85-88%.
1. A failing startup takes longer to reach peak cpu than a successful startup.
1. CPU usage will sit at 71-73% cpu for 2-3 seconds before dropping to ~1% to hang.

**Comment by ogrotten** [comment]
[Discovered an ini entry placement and spelling issue](https://github.com/TanninOne/modorganizer/issues/501). Without the offending line in the wrong ini, startup viability increases to about 60%.

**Comment by ogrotten** [comment]
New troubleshooting

* Looked at the Windows Error Report. Saw that the last dll listed was from AMD Raptr. Uninstalling it seemed to fix the above problem.
* Setup F4SE, the above problem returned. Removed F4SE the problem remained.
* Un/reinstalled F4 again (about the 5th time now, yay google fiber) and MO 202 and 203.
* Setup MO with the default 3-line MO.ini set to read only. Problem gone for 5 consecutive runs.

To verify that a MO.ini entry may be the culprit . . . 
* I did the manual setup for enabling mods and manually installed *Homemaker - Expanded Settlements*. When started from the vanilla Launcher, the game ran fine.
* Installed *Homemaker - Expanded Settlements* via MO. Ran the game thru MO with the 3-line MO.ini set to read only. Game starts, settlement editor crashes.
* Enabled writing on the MO.ini . . . well it worked (ish) this time. (Earlier today, it would work on the first launch, but then go back to the hanging behaviour.) The game still crashes when entering settlement editor.

**Comment by DeadMor0z** [comment]
Check if Homemeaker is installed correctly in MO. I have a problem with it, when installing via MO - it puts meshes and textures folders inside Data folder, not in mod root. When this happens - game crashes on save load when there are homemaker's objects nearby, or when entering settlement editor mode. If you move meshes and textures folders in mod root - everything is fine. This happens only with homemaker, bad fomod script, probably.

**Comment by ogrotten** [comment]
@DeadMor0z hm that would mean that the game worked with my manual install for some other reason.

Will check.

**Comment by ogrotten** [comment]
Well no amount of fixing on Homemaker makes it work with either 202 or 203. which is fucking stupid as a week ago it was working on 202.

**Comment by ogrotten** [comment]
I think I've pinpointed the problem here. 

With (I think) 203a, I setup the ini as outlined in #486.

* When I leave it alone, this inevitably leads to the game hanging while loading. The 1/10 times that it doesn't hang, it will ctd when entering the workshop.
* If I set the ini to read-only, MO complains about it being read-only, but then operates normally. The game will not hang and the workshop works without a hitch.

Clearly, something is being written to the ini that MO does't want to read.

I may go thru copying over 1 line at a time until it crashes.

**Comment by ogrotten** [comment]
Well It turns out this isn't the panacea that I thought it was. 

When I first opened this issue, with no mods installed, I had about a 1 of 10 success rate starting the game and then was unable to use the settlement editor.

* If I allow MO 203 to build a full ini file and allow it write access, this success rate is still the case. With or without mods. 
* If I setup the ini so that it doesn't crash on first start, and make it read-only, the success rate rises to about 2-3 of 5 and I can then use the settlement editor. With or without mods.

In all cases, the game starts and runs perfectly well without MO.

I've done all the detective work I can handle here. I'm happy to answer questions if need be.

**Comment by DeadMor0z** [comment]
What windows version do you run it on?

**Comment by ogrotten** [comment]
@DeadMor0z Wind 10

**Comment by ogrotten** [comment]
v2.0.4 still hangs on load most of the time. 

* Fresh install of the game.
* Fresh install of all mods from archive.

However, something is different. Previously, it would hang in the 650-680 MB used range. Now it hangs in the 430-440 MB used range. . . sometimes.

So far, it works 1 of 5-6 times. When it doesn't work, it hard locks at the above stated range. and I have to use task manager.

**Comment by ogrotten** [comment]
I discovered that I had Homemaker and a couple others were out of order. Putting them in the right place seems to have fixed the problem for tonight. Will report back if I notice otherwise.

**Comment by ogrotten** [comment]
This is still a problem. It will run about 1 of 5 times. It will hang the other 4.

I have, however, discovered that the hang is EXTREMELY consistent. It will hang between memory usage of 437.4 and 438.4 MB used every time. This hang is downline of slow loading. It will hang at this point after it has taken about twice as long to this point as a successful run.

**Comment by ogrotten** [comment]
This is still a problem. It will run about 1 of 5 times. It will hang the other 4.

I have, however, discovered that the hang is EXTREMELY consistent. It will hang between memory usage of 437.4 and 438.4 MB used every time.

**Comment by ogrotten** [comment]
Is there somewhere or some way that I can force come kind of verbose mode of either Fallout or MO so that I can get some kind of report?

Any suggestions on what I might check here? I'm not truly convinced of any facet of this problem.

**Comment by ogrotten** [comment]
To reiterate and make clear: 

* Without any version of MO, Fallout 4 runs without problem.
* With version a2 or later of MO, Fallout 4 will hang during boot 2-4 out of 5 starts. 
* I have performed many troubleshooting steps including reinstall of both Fallout 4 and MO, to moving both to a different hard drive.
* On version a4, 98% of the time the hang occurs during load at 438 megabytes used. ~50% of the time it is 438.6mb used.

The game will run if I end task and restart it enough times. This is what I'm doing in order to play the game. I just do it over and over till it works.

**Comment by Hullaballoonatic** [comment]
I'd like to join and share I have the same problem. I'll expand on how I experience it, but starting with features maybe unique to my setup:
- 4670k @4200 MHz
- 16GB DDR3 (25000MB allocated with ENBoost)
- 970
- Win10
- MO2A4 is installed to my main SSD, while Fallout4 and MO mods are on a different SSD

My experience:
- I launch the game, F4SE loads without problem (potentially, read below)
- The Fallout 4 exe launches with a black screen, and the windows recording feature displays a prompt. CPU usage is at about ~35%
- There's a short grey screen just before the active window switches to the full screen executable, and there's a short and miniature spike of CPU usage to ~40%
- The screen is black, and ENB displays it has loaded correctly
- Simultaneously, if the game hangs, CPU usage plummets immediately to <1%, otherwise it rises above 40% and I can play
- Success rate ~1/10

The F4SE thing (potentially related):
- Plugins I do not have added to the Virtual File Tree are still somehow loaded.
- I'm certain the plugin files are not in my vanilla f4 folder or reported in the data breakdown within MO.
- F4SE has never failed to load regardless.
- I have tried reinstalling F4SE with no effect.
- This isn't causing me a problem at the moment.

**Comment by Hullaballoonatic** [comment]
A further update. I haven't done definitive testing to know if this description is absolutely correct, but it appears that if i do no changes to my files then the success rate is near 100%, and the more i mess with them between attempts, the less likely it is to succeed.

**Comment by ogrotten** [comment]
I've noticed that mine is fairly predictable.

If i start the game, no MO, manual or NMM mod installation OR without mods completely, it works 100%

When I use MO, start game hang isn't as random as I thought. 
- Starts 1-4 fail
- Start 5 succeeds
- Start 6 thru 9-11 fail.
- One successful start.
- Fewer fails
- etc., until . . . 

. . . until failures are reduced to a point where it doesn't fail at all. I BELIEVE that this is between starts of MO which would mean that it depends on whatever caching by the OS.

**Comment by Hullaballoonatic** [comment]
It doesn't always hang at the same place every time. Sometimes it hangs before windows makes it the active window, sometimes after, and sometimes it hangs at the steam bootstrapper for me (that was a new experience for me this morning). it SEEMS to improve between relaunching MO, but i can't absolutely verify that. at least it more consistently reaches the further stages of the full game launch the more repeated attempts i make.

unfortunately it's not consistent day to day or even hour to hour the absolute behavior. either this means that changes to the OS is affecting it and/or my own changes to the mod loadout is. I can't pinpoint it.

I wonder if the two of us could try to narrow down similarities between us to help diagnosis? Clearly very few people have this issue, but its horrible.

**Comment by ogrotten** [comment]
I have a "BLANK" profile. All the mods, no checks. Starting F4 from that profile has better behaviour as I've outlined above, but it still fails more often than not.

It may just be that it doesn't like the VFS module.

@TanninOne it's been a LONG while since, but back in my Quake 2 days, a mod author was having a stupid side-effect problem in game, caused by his mod... pickups would become ungettable. You'd run over it and it wouldn't pickup. After many weeks of trying to figure out what the hell was the problem, someone suggested that he use a different compiler. He did so and the problem went away (with better performance).

Therefore, i'm suggesting that you may consider using a different compiler.

**Comment by Hullaballoonatic** [comment]
Baffling. Ever since the new patch came out, it's booted without stalling every time. Of note, I updated Shadow Boosts corresponding patch, but haven't had F4SE updated yet, so am loading outside it. I also haven't made any particular changes to my load order or VFT, and finally, perhaps of very poignant note: the mods menu in the fallout main menu shows no mods, but they are definitely present. Maybe that's unrelated.

**Comment by Blze001** [comment]
In my variation of this bug (seems like it's popping up in different ways for different people) the issue seems to stem from trying to launch Fallout4.exe without Steam already running. When this happens, even if you close MO2, start Steam, and reopen MO2, it will still hang when Fallout4.exe is still a background task only using 1.6MB. I have to sign out of my user account and sign back in to resolve it.

**Comment by emilykmb** [comment]
Same issue- mine started happening after I installed Automatron, which I initially thought was the culprit but I now believe that it was simply the size of the dlc on top of the mods. My best guess is that it can't load all the data quickly enough sometimes, but I have no clue how to improve it other than drastically reducing my load order.

**Comment by ogrotten** [comment]
Since my last comment on Feb 26 2016, the problem has not changed. The parameters of when it hangs have changed a bit . . . the memory range is now different and is likely relevant to the number of mods I've installed. However, the times it hangs are the same.

It occurred to me the other day that If I've rebooted, start MO, and then start trying to run the game many times in succession, it will start in as few as 2 tries.

It seems to have something to do with caching. It's as if the data is being read or stored where it causes the game to hang, but when being cached that error is corrected. 

I say cached because I watch the amount of memory used each start. For a hang, memory usage will step slowly thru the 200-450mb used range while it reads from disk. When it does NOT hang, memory usage will zip from < 200 up past 600mb used and the load screen will start. 

Also, those times that a successful start comes at a surprising moment, such as the 2nd try, it always goes to one of the SPECIAL videos. I dunno how much that is relevant, but it's an observation.

**Comment by ogrotten** [comment]
I haven't played the game since probably the last time I posted to this thread. 

Probably due to the update, 204 now doesn't work at all. The hang is persistent in the ~450mb used range. At it's worst, it used to be 1 start in 10 tries... I tried ~30 times (lost count) and it didn't start once.

Went back to 202 which worked the first time, and then not ever again. So I have no fucking clue.

On a secondary note, i don't see anywhere a link to your Patreon page. Will start a new issue.
